tests/unit/models/server-watcher-test.js

```diff
@@ -17,10 +17,17 @@ describe('Server Watcher', function () {
     analytics = new MockAnalytics();
     watcher = new MockServerWatcher();
 
-    await ServerWatcher.build({
+    class ServerWatcherMock extends ServerWatcher {
+      setupBroccoliWatcher() {
+        this.watcher = watcher;
+
+        return super.setupBroccoliWatcher(...arguments);
+      }
+    }
+
+    await ServerWatcherMock.build({
       ui,
       analytics,
-      watcher,
```

Comment:
Assigning the `watcher` upon creation is not really how things work at the moment, and is why this test didn't catch `this.watcher` being `undefined` in the server-watcher's `constructor` when running `ember s` in a real project. This test now fails without the changes in this PR and succeeds with.